src/compilation/config.rs

```diff
@@ -0,0 +1,128 @@
+use crate::{
+    content_encoding::{
+        ContentTypeCheckType, ContentTypeConverterType, DEFAULT_CONTENT_ENCODING_CHECK,
+        DEFAULT_CONTENT_ENCODING_CONVERT,
+    },
+    media_type::{MediaTypeCheckType, DEFAULT_MEDIA_TYPE_CHECK},
+    schemas,
+};
+use serde_json::Value;
+use std::{collections::HashMap, fmt};
+
+#[allow(missing_docs)]
+#[derive(Clone, Default)]
+pub struct JSONSchemaCompileConfig {
+    draft: Option<schemas::Draft>,
+    custom_content_media_type_checks: HashMap<String, Option<MediaTypeCheckType>>,
+    custom_content_encoding_checks: HashMap<String, Option<ContentTypeCheckType>>,
```

Comment:
What is the case for having `None` as a possible value in these hashmaps?

Reply:
This would require some documentation ... and I'm sorry that I've not done it yet.

The idea is that whenever we're getting a a content encoding check we're giving priority to the "user-defined" one first, if we haven't found it then we use the defaults.

The idea of allowing `None` in there is to allow users to "de-register" default formats.

Using this approach in order to have a static and immutable representation of the default handled mediaTypes, contentEncoding (and in the future formats) to reduce as much as possible the memory impact in case many many JSONSchema instances (or configs) are created